pkg/database/openshiftclusters.go

```diff
@@ -120,6 +124,41 @@ func (c *openShiftClusters) Get(ctx context.Context, key string) (*api.OpenShift
 	}
 }
 
+// QueueLength returns OpenShiftClusters un-queued document count.
+// If error occurs, 0 is returned with error message
+func (c *openShiftClusters) QueueLength(ctx context.Context) (int, error) {
+	partitions, err := c.collc.PartitionKeyRanges(ctx, "OpenShiftClusters")
+	if err != nil {
+		return 0, err
+	}
+
+	var countTotal int
+	for _, r := range partitions.PartitionKeyRanges {
+		result := c.c.Query("", &cosmosdb.Query{
+			Query: `SELECT VALUE COUNT(1) FROM OpenShiftClusters doc WHERE doc.openShiftCluster.properties.provisioningState IN ("Creating", "Deleting", "Updating") AND (doc.leaseExpires ?? 0) < GetCurrentTimestamp() / 1000`,
+		}, &cosmosdb.Options{
+			PartitionKeyRangeID: r.ID,
+		})
+		// because we aggregate count we don't expect pagination in this query result,
+		// so we gonna call Next() only once per partition.
+		var data struct {
+			api.MissingFields
+			Document []string `json:"Documents,omitempty"`
+		}
+		err := result.NextRaw(ctx, &data)
```

Comment:
I don't understand how this works.  Trying the query manually I get the result `[0]` - how will that deserialise into data?

Reply:
I think it might be becaus you dont have any docoments?
```{"Metric":"database.openshiftclusters.queue.length","Account":"*","Namespace":"*","Dims":{"hostname":"unknown","location":"eastus"},"TS":"2020-01-24T11:38:55.679"}:0|g
{"Metric":"database.openshiftclusters.queue.length","Account":"*","Namespace":"*","Dims":{"hostname":"unknown","location":"eastus"},"TS":"2020-01-24T11:40:35.324"}:1|g
```
Will try to apply some dummy docs to re-verify

Reply:
Here is where I did some loadtesting:
```
{"Metric":"database.openshiftclusters.queue.length","Account":"*","Namespace":"*","Dims":{"hostname":"unknown","location":"eastus"},"TS":"2020-01-24T11:58:38.631"}:9|g
```

Reply:
TIL moment for me.